src/acceptance-test/java/uk/gov/ida/verifyserviceprovider/SuccessMatchAcceptanceTest.java

```diff
@@ -170,4 +175,147 @@ public void shouldReturnAnErrorWhenInvalidEntityIdProvided() {
         assertThat(errorMessage.getCode()).isEqualTo(BAD_REQUEST.getStatusCode());
         assertThat(errorMessage.getMessage()).isEqualTo("Provided entityId: invalidEntityId is not listed in config");
     }
+
+    @Test
+    public void shouldReturnAnErrorWhenNullRequestIdIsProvided() {
+        RequestResponseBody requestResponseBody = generateRequestService.generateAuthnRequest(application.getLocalPort());
+
+        Map<String, String> translateResponseRequestData = new HashMap<>(); // not a lot of map implementations allow nulls
```

Comment:
Is this comment saying HashMap does allow nulls?  Is that a null value or a null key or both?  I'd understand if it was some obscure `ConcurrentFooBarRoboMap` or something as then the comment would imply the choice of Map implementation was because it accepted nulls but as `HashMap` is the go-to general purpose map, I don't think this comment is adding anything of value and actually detracts from the readability of the code (I literally can't concentrate on anything else because I'm wondering what it's telling me that I don't understand).  And this is coming from someone who isn't dogmatic about "no comments".  What am I missing?

Reply:
I was explaining why I hadn't used the same pattern that appears in the all the existing tests - because `HashMap` seems to be just about the only implementation of Map that allows a null value.

Reply:
More than happy to remove.

Reply:
I'd committed the sin of not reviewing the entire file.  However, if it had said `// ImmutableMap does not allow null values` then I'd have probably known what was going on without reading the whole file.  😉

Reply:
Updated - no worries. Comment wasn't really necessary.